### ISSUE TITLE
Autoflush is not working after parts of entities deleted

### ISSUE THREAD

**Issue by del-zhenwu** [milvus-io/milvus#3653]
**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
delete cmd:
```
locust -f search_task.py -u 1000 -r 10 --step-load --step-users 10 --step-time 1m
```
`search_task.py`
```
    @task(1)
    def delete(self):
        delete_ids = random.sample(ids, 100)
        logging.error(delete_ids)
        self.client.delete(delete_ids)
```

The interval of `flush` is about 8min
```
[2020-09-08 14:57:28,967][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Count: 286, Usage: 3746MB, Capacity: 4096MB
[2020-09-08 15:05:18,811][DEBUG][SERVER][PrintTimeRecord][timing_flush] SegmentWriter::WriteDeletedDocs: /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1570/F_6753: totally cost (0.022797 second [22.796502 ms])
[2020-09-08 15:05:18,831][DEBUG][SERVER][PrintTimeRecord][timing_flush] SegmentWriter::WriteBloomFilter: /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1570/F_6754: totally cost (0.000003 second [0.003118 ms])
[2020-09-08 15:05:19,716][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Insert /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1571/F_6655 size: 0MB into cache
[2020-09-08 15:05:19,717][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Count: 287, Usage: 3746MB, Capacity: 4096MB
[2020-09-08 15:12:59,965][DEBUG][SERVER][PrintTimeRecord][timing_flush] SegmentWriter::WriteDeletedDocs: /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1571/F_6755: totally cost (0.024119 second [24.118794 ms])
[2020-09-08 15:12:59,966][DEBUG][SERVER][PrintTimeRecord][timing_flush] SegmentWriter::WriteBloomFilter: /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1571/F_6756: totally cost (0.000004 second [0.004253 ms])
[2020-09-08 15:13:00,926][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Insert /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1572/F_6657 size: 0MB into cache
[2020-09-08 15:13:00,928][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Count: 288, Usage: 3746MB, Capacity: 4096MB
[2020-09-08 15:20:37,491][DEBUG][SERVER][PrintTimeRecord][timing_flush] SegmentWriter::WriteDeletedDocs: /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1572/F_6757: totally cost (0.078353 second [78.353018 ms])
[2020-09-08 15:20:37,496][DEBUG][SERVER][PrintTimeRecord][timing_flush] SegmentWriter::WriteBloomFilter: /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1572/F_6758: totally cost (0.000013 second [0.012994 ms])
[2020-09-08 15:20:38,943][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Insert /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1573/F_6659 size: 0MB into cache
[2020-09-08 15:20:38,949][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Count: 289, Usage: 3746MB, Capacity: 4096MB
[2020-09-08 15:28:20,787][DEBUG][SERVER][PrintTimeRecord][timing_flush] SegmentWriter::WriteDeletedDocs: /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1573/F_6759: totally cost (0.217214 second [217.214103 ms])
[2020-09-08 15:28:20,787][DEBUG][SERVER][PrintTimeRecord][timing_flush] SegmentWriter::WriteBloomFilter: /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1573/F_6760: totally cost (0.000016 second [0.015551 ms])
[2020-09-08 15:28:22,244][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Insert /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1574/F_6661 size: 0MB into cache
[2020-09-08 15:28:22,246][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Count: 290, Usage: 3746MB, Capacity: 4096MB
[2020-09-08 15:36:11,736][DEBUG][SERVER][PrintTimeRecord][timing_flush] SegmentWriter::WriteDeletedDocs: /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1574/F_6761: totally cost (0.028275 second [28.274568 ms])
[2020-09-08 15:36:11,751][DEBUG][SERVER][PrintTimeRecord][timing_flush] SegmentWriter::WriteBloomFilter: /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1574/F_6762: totally cost (0.000013 second [0.012752 ms])
[2020-09-08 15:36:12,704][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Insert /test/milvus/db_data_011/sift_10m_100000_128_l2_delete/db/collections/C_6/P_6/S_1575/F_6663 size: 0MB into cache
[2020-09-08 15:36:12,704][DEBUG][SERVER][insert_internal][timing_flush] [CACHE CPU] Count: 291, Usage: 3746MB, Capacity: 4096MB

```

**Steps/Code to reproduce behavior**
deleting test based on sift-10m, ids ranges from 0 to 9999999
the count entities of collection is not changed

```
>>> from milvus import Milvus
>>> m=Milvus(host="192.168.1.112", port=19532)
>>> m.count_entities('sift_10m_100000_128_l2')
9349701
>>> m.count_entities('sift_10m_100000_128_l2')
9349701
>>> m.count_entities('sift_10m_100000_128_l2')
9349701
>>> m.count_entities('sift_10m_100000_128_l2')
9349701
>>> m.count_entities('sift_10m_100000_128_l2')
9349701
```

**Expected behavior**

**Method of installation**
- [x] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)

- Milvus version (master or released version)

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**

**Additional context**

**Comment by yhmo** [comment]
Hangs in MemCollection::ApplyDeleteToFile()

**Comment by yhmo** [comment]
Fixed by #3667

**Comment by del-zhenwu** [comment]
close
